benchmark/benchmarks.jl

```diff
@@ -0,0 +1,12 @@
+# Each file of the form "bench_$(name).jl" in this directory is `include`d and
+# its last statement is assumed to be a `BenchmarkGroup`.  This group is added
+# to the top-level group `SUITE` with the `$name` extracted from the file name.
```

Comment:
What do you think about this strategy? I use this for all my packages and it works well for me.  This is useful for running each file separately like this: `run(include("benchmark/bench_qr.jl"))`.

Reply:
This sounds completely fine to me!

---

PS: My fantasy interface might be more akin to `@testset` where the groups are assembled "magically" using dynamic scope

```julia
@benchmark_group "all" begin
    @benchmark_group "QR $K" for K=1:22
        a = rand(SMatrix{K,K,Float64,K*K})
        m = Matrix(a)
        @benchmarkable qr($a)
        @benchmarkable qr($m)
     end
     # other groups  or includes()...
end
```

but unlike `Test`, to allow running these selectively. But no doubt that's an issue for BenchmarkTools ;-)

Reply:
> PS: My fantasy interface might be more akin to @testset where the groups are assembled "magically" using dynamic scope

Ref https://github.com/JuliaCI/PkgBenchmark.jl/issues/9#issuecomment-257952067.

Reply:
Thanks for the crossref :+1:

I do strongly disagree with Jarret on that one. I think dynamic scoping (not global scope mind you!) can be fairly clear and non-magical for this kind of use as long as it's clearly documented.